src/ECS/System/TestTeleportingDeerSystem.cpp

```diff
@@ -5,17 +5,22 @@ namespace Sourcehold {
         namespace System {
             void TestTeleportingDeerSystem::tickEntity(entt::registry &registry, EachEntityType &entity) {
                 auto [entityRef, typeComponent, positionComponent] = entity;
-                if (typeComponent.type != EntityType::DEER) return;
+                if (typeComponent.type != EntityType::WOODCUTTER) return;
 
-                // just a test, remove or optimize in future
-                std::random_device dev;
-                std::mt19937 rng(dev());
-                std::uniform_int_distribution<std::mt19937::result_type> dist(0, 2);
-
-                registry.emplace_or_replace<Component::Position>(entityRef,
-                    positionComponent.x + int(dist(rng)) - 1,
-                    positionComponent.y + int(dist(rng)) - 1
-                );
+                auto view = registry.view<Component::Position, Component::EntityType>();
+                for(auto anyEntity: view.each()) {
+                    auto [ anyEntityRef, anyPositionComponent, anyTypeComponent] = anyEntity;
+                    if (anyTypeComponent.type != EntityType::TREE_CHESTNUT_XL) continue;
+                    // registry.emplace_or_replace<Component::Position>(entityRef,
+                    //     anyPositionComponent.x < positionComponent.x ? positionComponent.x - 1 : positionComponent.x + 1,
+                    //     anyPositionComponent.y < positionComponent.y ? positionComponent.y - 1 : positionComponent.y + 1
+                    // );
+                    registry.emplace_or_replace<Component::MoveTarget>(entityRef,
+                        false,
+                        Component::Position { anyPositionComponent.x, anyPositionComponent.y }
+                    );
+                    break;
+                }
             }
         }
     }
```

Comment:
If you plan to add more tests, consider using [Google Test](https://github.com/google/googletest).